### PR TITLE
[RDR] Enable ACM observability for DR monitoring dashboard on RHACM console

### PR DESCRIPTION
**Pull request by am-agrawa**
Whitelist RDB metrics and enable ACM observability to access DR monitoring dashboard on the RHACM console.

Also fixes https://github.com/red-hat-storage/ocs-ci/issues/9622

### CODE REVIEW

File: ocs_ci/deployment/deployment.py [hunk 1]
```diff
@@ -2877,6 +2883,80 @@ def deploy(self):
...
+            "oc get MultiClusterObservability observability -o jsonpath='{.status.conditions[1].status}'"
+        )
+        if check_observability_status:
+            logger.info("ACM observability is successfully enabled")
+        else:
+            raise ACMObservabilityNotEnabled(
+                "ACM Observability is not enabled, status is False"
+            )
+
+    def enable_acm_observability(self):
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py:2938]: This function calls/creates other yamls needed for observability and is not being called elsewhere. Shouldn't that be fine?

File: ocs_ci/templates/DR/multiclusterobservability.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   interval: 300
+ storageConfig:
+   alertmanagerStorageSize: 1Gi
+   compactStorageSize: 100Gi
+   metricObjectStorage:
+     key: thanos.yaml
+     name: thanos-object-storage
+   receiveStorageSize: 100Gi
+   ruleStorageSize: 1Gi
+   storageClass:  thin-csi-odf
```
Comment (prsurve) [inline ocs_ci/templates/DR/multiclusterobservability.yaml:21]: This value should be configurable

File: ocs_ci/deployment/deployment.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py:3017]: done

File: ocs_ci/templates/DR/multiclusterobservability.yaml (hunk 2 above)
Comment (am-agrawa) [reply ocs_ci/templates/DR/multiclusterobservability.yaml]: done

File: ocs_ci/ocs/constants.py [hunk 3]
```diff
@@ -2257,7 +2257,6 @@
     "&rows_per_page=25&delta=1296000&contains=submariner-operator-bundle-container-v"
 )
 SUBMARINER_BREW_REPO = "brew.registry.redhat.io/rh-osbs/iib"
-SUBCTL_DOWNSTREAM_URL = "registry.redhat.io/rhacm2/"
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/ocs/constants.py]: added back

File: ocs_ci/deployment/deployment.py [hunk 4]
```diff
@@ -2938,6 +2947,110 @@ def deploy(self):
...
+        Create a bucket to be used in the thanos.yaml for ACM observability
+        Args:
+            acm_indexes (list): List of acm indexes
+        """
+        self.bucket_name = ""
+        for index in acm_indexes:
+            self.bucket_name += config.clusters[index].ENV_DATA["cluster_name"]
+        return self.bucket_name
+
+    retry(CommandFailed, tries=10, delay=30)
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: no, fixed it

File: ocs_ci/deployment/deployment.py [hunk 5]
```diff
@@ -40,6 +40,9 @@
     get_lvm_full_version,
     check_cephcluster_status,
 )
+from ocs_ci.ocs.constants import (
+    OBSERVABILITYMETRICSCONFIGMAP_PATH,
+)
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: fixed

**Comment by am-agrawa** [comment]
Passed here- https://url.corp.redhat.com/d200c31

**Review by prsurve — APPROVED** [review]

**Review by prsurve — APPROVED** [review]

**Review by keesturam — DISMISSED** [review]

File: ocs_ci/deployment/deployment.py [hunk 6]
```diff
@@ -3444,6 +3449,105 @@ def deploy(self):
         else:
             self.enable_managed_serviceaccount()
 
+    @retry(ACMObservabilityNotEnabled, tries=10, delay=30)
+    def check_observability_status(self):
+        """
+        Check observability status
+        Returns (bool): True or False
```
Comment (ebondare) [inline ocs_ci/deployment/deployment.py:3456]: It doesn't return True or False, it raises ACMObservabilityNotEnabled. Which logic did you actually choose?

File: ocs_ci/deployment/deployment.py [hunk 7]
```diff
@@ -3444,6 +3449,105 @@ def deploy(self):
...
+        acm_observability_status = bool(
+            run_cmd(
+                "oc get MultiClusterObservability observability -o jsonpath='{.status.conditions[1].status}'"
+            )
+        )
+
+        if acm_observability_status:
+            logger.info("ACM observability is successfully enabled")
+        else:
+            logger.error("ACM observability could not be enabled, re-trying...")
```
Comment (ebondare) [inline ocs_ci/deployment/deployment.py:3469]: Did you plan to add some code for re-trying?

File: ocs_ci/deployment/deployment.py (hunk 6 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py:3456]: thanks, it was a stale doctext, fixed it

File: ocs_ci/deployment/deployment.py (hunk 7 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py:3469]: @retry decorator is already added to the function

**Review by prsurve — DISMISSED** [review]

File: ocs_ci/deployment/deployment.py [hunk 8]
```diff
@@ -3444,6 +3449,106 @@ def deploy(self):
...
+        thanos_secret_data["stringData"]["thanos.yaml"] = str(navigate_thanos_yaml)
+        thanos_data_yaml = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="thanos", delete=False
+        )
+        templating.dump_data_to_temp_yaml(thanos_secret_data, thanos_data_yaml.name)
+
+        logger.info(
+            "Creating thanos.yaml needed for ACM observability after passing required params"
+        )
+        run_cmd(f"oc create -f {thanos_data_yaml.name}")
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/deployment.py:3505]: run_cmd is a deprecated form of exec_cmd, we need to change replace it here

File: ocs_ci/deployment/deployment.py [hunk 9]
```diff
@@ -3444,6 +3449,106 @@ def deploy(self):
...
+    def check_observability_status(self):
+        """
+        Check observability status
+
+        raises ACMObservabilityNotEnabled: if the cmd returns False, ACM observability is not enabled
+
+        """
+
+        acm_observability_status = bool(
+            run_cmd(
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/deployment.py:3462]: run_cmd is a deprecated form of exec_cmd, we need to change replace it here

File: ocs_ci/deployment/deployment.py [hunk 10]
```diff
@@ -3444,6 +3449,106 @@ def deploy(self):
...
+        ] = defaultstorageclass[0]
+        multiclusterobservability_data_yaml = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="multiclusterobservability", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            multiclusterobservability_yaml_data,
+            multiclusterobservability_data_yaml.name,
+        )
+
+        run_cmd(f"oc create -f {multiclusterobservability_data_yaml.name}")
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/deployment.py:3537]: run_cmd is a deprecated form of exec_cmd, we need to change replace it here

File: ocs_ci/deployment/deployment.py [hunk 11]
```diff
@@ -3444,6 +3449,106 @@ def deploy(self):
...
+        logger.info("Create thanos secret yaml")
+        self.thanos_secret()
+
+        logger.info("Whitelist RBD metrics by creating configmap")
+        run_cmd(f"oc create -f {constants.OBSERVABILITYMETRICSCONFIGMAP_PATH}")
+
+        logger.info(
+            "Add label for cluster-monitoring needed to fire VolumeSyncronizationDelayAlert on the Hub cluster"
+        )
+        run_cmd(
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/deployment.py:3548]: run_cmd is a deprecated form of exec_cmd, we need to change replace it here

File: ocs_ci/deployment/deployment.py [hunk 12]
```diff
@@ -3444,6 +3449,106 @@ def deploy(self):
...
+            multiclusterobservability_data_yaml.name,
+        )
+
+        run_cmd(f"oc create -f {multiclusterobservability_data_yaml.name}")
+
+        logger.info("Create thanos secret yaml")
+        self.thanos_secret()
+
+        logger.info("Whitelist RBD metrics by creating configmap")
+        run_cmd(f"oc create -f {constants.OBSERVABILITYMETRICSCONFIGMAP_PATH}")
```
Comment (DanielOsypenko) [inline ocs_ci/deployment/deployment.py:3543]: run_cmd is a deprecated form of exec_cmd, we need to change replace it here

File: ocs_ci/deployment/deployment.py (hunk 8 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

File: ocs_ci/deployment/deployment.py (hunk 9 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

**Review by DanielOsypenko — DISMISSED** [review]

File: ocs_ci/deployment/deployment.py (hunk 10 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

File: ocs_ci/deployment/deployment.py (hunk 11 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

File: ocs_ci/deployment/deployment.py (hunk 12 above)
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

File: ocs_ci/deployment/deployment.py [hunk 13]
```diff
@@ -3444,6 +3449,106 @@ def deploy(self):
         else:
             self.enable_managed_serviceaccount()
 
+    @retry(ACMObservabilityNotEnabled, tries=10, delay=30)
+    def check_observability_status(self):
+        """
+        Check observability status
+
+        raises ACMObservabilityNotEnabled: if the cmd returns False, ACM observability is not enabled
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/deployment/deployment.py]: done

**Review by ebondare — APPROVED** [review]

**Review by keesturam — APPROVED** [review]

**Pull request merged**